src/random_access_file.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ /** @file random_access_file.cpp Actual implementation of the RandomAccessFile class. */
+
+#include "stdafx.h"
+#include "random_access_file_type.h"
+
+#include "debug.h"
+#include "fileio_func.h"
+#include "string_func.h"
+
+#include "safeguards.h"
+
+/**
+ * Create the RandomAccesFile.
+ * @param filename Name of the file at the disk.
+ * @param subdir   The sub directory to search this file in.
+ */
+RandomAccessFile::RandomAccessFile(const std::string &filename, Subdirectory subdir) : filename(filename)
+{
+	this->file_handle = FioFOpenFile(filename, "rb", subdir);
+	if (this->file_handle == nullptr) usererror("Cannot open file '%s'", filename.c_str());
+
+	/* When files are in a tar-file, the begin of the file might not be at 0. */
+	long pos = ftell(this->file_handle);
+	if (pos < 0) usererror("Cannot read file '%s'", filename.c_str());
+
+	/* Store the filename without path and extension */
+	auto t = filename.rfind(PATHSEPCHAR);
+	std::string name_without_path = filename.substr(t != std::string::npos ? t + 1 : 0);
+	this->simplified_filename = name_without_path.substr(0, name_without_path.rfind('.'));
+	strtolower(this->simplified_filename);
+
+	this->SeekTo((size_t)pos, SEEK_SET);
+}
+
+/**
+ * Close the file's file handle.
+ */
+RandomAccessFile::~RandomAccessFile()
+{
+	fclose(this->file_handle);
+}
+
+/**
+ * Get the filename of the opened file with the path from the SubDirectory and the extension.
+ * @return Name of the file.
+ */
+const std::string& RandomAccessFile::GetFilename() const
```

Comment:
```suggestion
const std::string &RandomAccessFile::GetFilename() const
```
(and others) no?

Reply:
There was one other. I actually noticed MSVC messing things back when I wrote it, so I disabled MSVC's messing. However, overlooked these two cases.